### PR TITLE
Move character stat names into an enum

### PR DESCRIPTION
**Pull request by CJPoll**
Instead of having class attributes, just have an enum of stat
identities.

### CODE REVIEW

**Comment by Rybec** [comment]
Suggestion:

```
    self._baseStats = {
```
-           self.BODY_STAT_NAME : self.STARTING_STAT_VALUE,
-           self.MIND_STAT_NAME : self.STARTING_STAT_VALUE,
-           self.SOUL_STAT_NAME : self.STARTING_STAT_VALUE
-           Stats.BODY : self.STARTING_STAT_VALUE,
-           Stats.MIND : self.STARTING_STAT_VALUE,
-           Stats.SOUL : self.STARTING_STAT_VALUE
      }

Make it a list.  One benefit of an enum is that if it starts at 0 and is a complete series of values between 0 and the max, you can make it a list instead of a dict.

So something like this:
self._baseStats = [
    self.STARTING_STAT_VALUE,
    self.STARTING_STAT_VALUE,
    self.STARTING_STAT_VALUE,
]

Incidentally, if you added a "LAST_VALUE" or some such to the end of the enum (that is the total number of stats a unit needs), you could do this instead:

self._baseStats = [self.STARTING_STAT_VALUE for i in xrange(Stats.LAST_VALUE)]

(On a side note, typically variables in Python use underscores instead of camelcase.  The language does not enforce this convention, but any linter will probably expect it by default.)

**Comment by Rybec** [comment]
Oh, something important: If the stats enum is in a different file, you have to import it.  There are two options here:

import enums
If you use this you have to specify the enums namespace, for example enums.Stats.SOUL.

from enums import Stats
This will only import the Stats enum (better, because you probably don't need to import all of the enums), and it will import it into the local namespace, so you can just use Stats.SOUL.

I generally put the enums in the file with the class or classes that they are defined for, then when they need accessed by outside modules, I use the namespace of the module they are in.  Personally, I find this maintains organization better by keeping things close to their origins.  I am not saying you have to do this though.  If you find it works better for you to have an enums module that keeps the enums together, then go ahead.

**Comment by CJPoll** [comment]
@Rybec: In my next PR, I'm going to alter the code organization a bit. You're absolutely right about the structure; the easy thing is to make an `enums.py`, like I did this PR, but that's not terribly cohesive. What I'm going to do is create an `actor` subdirectory, which contains the classes & files for these actors (Actor, Stats, Equipment, and Inventory), including this enum.

So I agree that doing so is better code organization, but outside the scope of this PR. As such, could I get you to sign off on the review so I can merge?

File: src/character.py [hunk 1]
```diff
@@ -63,7 +59,7 @@ def injure(self, amount):
 		self._currentLife -= amount
 
 	def maxLife(self):
-		body = self.currentStatValue(self.BODY_STAT_NAME)
+		body = self.currentStatValue(Stats.BODY)
```
Comment (thebriz24) [inline src/character.py:62]: Maybe change the variable so a newcomer is guaranteed not to be confuse if this is referring to the stat named "body". Idiot proofing. Haha.

**Review by thebriz24 — COMMENTED** [review]
Optional: in interest of completeness write test for each stat. For example test_increase_base_stat is only testing body right now.

File: src/character.py (hunk 1 above)
Comment (CJPoll) [reply src/character.py:62]: @thebriz24 @thebriz Do you mean like calling it `body_stat`?

Comment (thebriz24) [reply src/character.py:62]: Haha. *Just realizes he's the idiot because it IS actually talking about the body stat. *

**Review by thebriz24 — APPROVED** [review]

**Pull request merged**